src/Client.ts

```diff
@@ -306,6 +327,7 @@ export default class Client {
   }
 
   private async ensureUserContactPublished(legacy = false): Promise<void> {
+    // TODO (VOODOO): this should do something, but currently does nothing for Voodoo contacts
```

Comment:
I would think as part of the "green pasture" approach, we'll want to keep this function as-is and create a parallel version in the VoodooManager or somewhere else fresh.

Making changes to these core methods is going to break our test suite and existing functionality

Reply:
It's a good callout. I was debating doing an opt-based conditional in each of the main methods or just writing a `VoodooClient`. Your opinion is a new client class would be better right?

Reply:
I do think that. Changing existing methods is going to be an endless task of fixing broken tests and keeping the type checker happy for little benefit in terms of learning.

I think a new client + the existing `apiClient` gets us the most learning the fastest.

Reply:
We'd have to copy over 2 methods from the old client to the new one `publishEnvelopes` + `listEnvelopes` and we should be able to test end to end.